test/Interop/Cxx/templates/Inputs/class-template-instantiation-errors.h

```diff
@@ -0,0 +1,22 @@
+#ifndef TEST_INTEROP_CXX_TEMPLATES_INPUTS_CLASS_TEMPLATE_INSTANTIATION_ERRORS_H
+#define TEST_INTEROP_CXX_TEMPLATES_INPUTS_CLASS_TEMPLATE_INSTANTIATION_ERRORS_H
+
```

Comment:
Would you mind adding the following:
```
template<class T>
 struct MagicWrapper {
   T t;
   int getValuePlusArg(T arg) const { return t.getValue() + arg.getValue(); }
 };
```

Reply:
Maybe also a nested enum or union type?

Reply:
I'm sorry, what exactly do you want to cover with that test? The difference between that and the template at the beginning of this file is that in your version we call `arg.getValue()`, right? Could you also elaborate on why nested enum/union types should be tested, what failure case do you envision?

Reply:
Not sure, but I'm worried a member that has a template argument might crash when getting imported as a generic class. When the function decl visitor goes looking for where the template param type it likely won't be able to find it. We should make sure we at least don't import that member (and make sure we don't crash).

As for nested enum/union types, I don't think that's as important to cover, but it is a case we should make sure we don't crash on that I don't think we currently have coverage for. Here's an example:
```
template<class T>
struct Parent {
  enum Foo : T { A, B, C };
};
```
Maybe just enums (not unions after all).

Reply:
Hm, I tried this and it seems we import Foo as a struct. I don't yet understand why, but I guess this can be addressed in a separate PR, as it's also related to typedeffed templates. SG?